src/Commands/Images/GenerateAbstract.php

```diff
@@ -0,0 +1,375 @@
+    {
+        $this->output = $output;
+
+        return parent::initialize($input, $output);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('fop:images:generate:' . static::IMAGE_TYPE)
+            ->setDescription('Regenerate ' . static::IMAGE_TYPE . ' thumbnails')
+            ->addArgument(
+                'format',
+                InputArgument::OPTIONAL,
+                'images formats comma separated',
+                'all'
+            )
+            ->addOption(
+                'delete',
+                'd',
+                InputOption::VALUE_OPTIONAL,
+                'Delete current thumbnails',
+                true
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $formats = explode(',', $input->getArgument('format'));
+        $delete = (bool) $input->getOption('delete');
+        $io = new SymfonyStyle($input, $output);
+
+        $sucess = $this->_regenerateThumbnails(static::IMAGE_TYPE, $delete, $formats);
+
+        if (!$sucess || count($this->errors)) {
+            $io->error('Unable to generate thumbnails');
+            $io->warning('The generation generate the folowing errors : ');
+            foreach ($this->errors as $error) {
+                $io->error($error);
+            }
+
+            return 1;
+        }
+
+        $io->success('Thumbnails generated with success for ' . static::IMAGE_TYPE);
+    }
+
+    /**
+     * The whole logic of this file comes from controllers/admin/AdminImagesController.php
+     * with some light adaptations
+     *
+     * @param string $type
+     * @param bool $deleteOldImages
+     * @param array $imagesFormats
+     *
+     * @return bool
+     */
+    protected function _regenerateThumbnails($type = 'all', $deleteOldImages = true, $imagesFormats = ['all'])
+    {
+        $languages = Language::getLanguages(false);
+        $process = [
+            ['type' => 'categories', 'dir' => _PS_CAT_IMG_DIR_],
+            ['type' => 'manufacturers', 'dir' => _PS_MANU_IMG_DIR_],
+            ['type' => 'suppliers', 'dir' => _PS_SUPP_IMG_DIR_],
+            ['type' => 'products', 'dir' => _PS_PROD_IMG_DIR_],
+            ['type' => 'stores', 'dir' => _PS_STORE_IMG_DIR_],
+        ];
+
+        // Launching generation process
+        foreach ($process as $proc) {
+            if ($type != 'all' && $type != $proc['type']) {
+                continue;
+            }
+
+            //Display which type currently processing
+            $this->output->writeln('Processing ' . $proc['type']);
+
+            // Getting format generation
+            $formats = ImageType::getImagesTypes($proc['type']);
+
+            if (!count($formats)) {
+                $this->output->writeln('<error>No format for ' . $proc['type'] . '</error>');
```

Comment:
No format for XXX should be displayed like others errors and should stop the process

Reply:
Should be good with last update

Reply:
Yes, it's ok now, thanks a lot @nenes25